bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckRun.java

```diff
@@ -1124,7 +1124,11 @@ private void checkStatus() {
             }
             PullRequestCheckIssueVisitor visitor = checkablePullRequest.createVisitor();
 
-            var version = BotUtils.getVersion(pr).orElse(null);
+            JdkVersion version = null;
+            try {
+                version = BotUtils.getVersion(pr).orElse(null);
+            } catch (Exception e) {
+            }
```

Comment:
I'm not sure ignoring an exception here is a good idea. In what situation did it throw one?

Reply:
> I'm not sure ignoring an exception here is a good idea. In what situation did it throw one?

It's because `BotUtils.getVersion` only works for fixVersion is configured in` .jcheck/conf`. If we use the override jcheck conf, it will throw exception. So it will make some override jcheck conf tests fail. I think we could fix it, but not in this pr.

Reply:
It needs to be solved in some way. Catching `Exception` and ignoring it is not allowed in bot code. That could be triggered by an IOException, or failure in contacting the remote or almost anything else. If that happens, we need to handle the failure correct so that the WorkItem will be retried.

BotUtils.getVersion either can't throw RuntimeException when it can't find a .jcheck/conf file, or we need to handle the fallback to the override repo. The latter would be cleaner, but we probably don't really need that support. I think returning Optional.empty() from BotUtils.getVersion() is probably ok if no .jcheck/conf file could be found.

Reply:
> It needs to be solved in some way. Catching `Exception` and ignoring it is not allowed in bot code. That could be triggered by an IOException, or failure in contacting the remote or almost anything else. If that happens, we need to handle the failure correct so that the WorkItem will be retried.
> 
> BotUtils.getVersion either can't throw RuntimeException when it can't find a .jcheck/conf file, or we need to handle the fallback to the override repo. The latter would be cleaner, but we probably don't really need that support. I think returning Optional.empty() from BotUtils.getVersion() is probably ok if no .jcheck/conf file could be found.

Got it! I will fix it

Reply:
Now I realized that this jcheck conf issue happens because the place of `JdkVersion version = BotUtils.getVersion(pr).orElse(null);`. In `CheckWorkItem`, we have check for jcheck configuration in **target branch**. So if jcheck conf missing or invalid in **target branch**, the bot will return immediately. 

However, in `BotUtils#getVersion`,  the bot would try to get conf from **source branch** first. Although we also have check for **source branch**, it is after the call of JdkVersion version = BotUtils.getVersion(pr).orElse(null);`. So bad conf in **source branch** would cause getVersion throw exceptions.